main.go

```diff
@@ -19,37 +19,48 @@ import (
 	"github.com/spf13/cobra"
 )
 
-func collectMetricsInBackground(config internal.ConfigStruct) {
+func collectMetricsInBackground(config internal.ConfigStruct, logger internal.ILogger) {
 	go func() {
 		client, err := internal.NewOtcClientFromConfig(config)
 		if err != nil {
-			panic(err)
+			logger.Panic("Error creating OTC client", "error", err)
 		}
 
+		logger.Info("New OTC Client created!")
+
 		var resourceIdToName map[string]string
 
 		if config.ResourceIdNameMappingFlag {
-			resourceIdToName, err = FetchResourceIdToNameMapping(client, config.Namespaces)
-			if err != nil {
-				panic(err)
+			resourceIdToName, err = FetchResourceIdToNameMapping(client, config.Namespaces, logger)
+			if err != nil || resourceIdToName == nil {
```

Comment:
Yes, I think I have those around the wrong way. In any case, I think this was something I added because my resource ID's were returning empty, but that was fixed with the env name fix, so this shouldn't be here any more.